_specifications/lsp/3.17/textDocument/didChange.md

```diff
@@ -2,6 +2,8 @@
 
 The document change notification is sent from the client to the server to signal changes to a text document. Before a client can change a text document it must claim ownership of its content using the `textDocument/didOpen` notification. In 2.0 the shape of the params has changed to include proper version numbers.
 
+Before requesting information from the server (e.g., `textDocument/completion` or `textDocument/signatureHelp`), the client must ensure that the document's state is synchronized with the server to guarantee reliable results.
```

Comment:
I think we need to make this more concrete since this sentences allows for interpretation as well (e.g. another change occurers before the client sends the request). So in a sequence

<document change one><code complete><document change two> the client needs to sync <document change one> to the server before requesting code compltion but not document change two.

Reply:
Sure. I added your example with a timeline table